### ISSUE TITLE
Many users reported that the upgrade process failed because files could not be replaced and caused NVDA corruption.

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#16122]
### Steps to reproduce:
Three patch versions have been pushed recently, which seems to have made this problem more obvious.
There are many reports from users that the upgrade cannot go smoothly because the files cannot be replaced, and even after the user clicks "retry" several times, the upgrade still fails and NVDA becomes corrupted.
After restarting the computer NVDA goes completely silent and they have to ask someone else or reinstall NVDA with the help of Narrator.

### Actual behavior:
The upgrade process failed because some files could not be replaced.
### Expected behavior:
Can be upgraded smoothly.
### NVDA logs, crash dumps and other attachments:
Due to the severity of the issue, users had to reinstall NVDA after their NVDA was compromised, which also resulted in log loss.
Since I have not encountered this problem myself, there is no log for the time being.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2023.3,2023.3.1,2023.3.2
#### Windows version:
Windows10 and Windows11
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Some users were able to upgrade successfully after restarting their computer, but others had their NVDA corrupted and unable to boot.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I've experienced this myself with other versions of NVDA, but it's become an issue too frequently lately.
#### If NVDA add-ons are disabled, is your problem still occurring?
Unknown at the moment
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Unknown at the moment

**Comment by josephsl** [comment]
Hi, when this happens again, press NVDA+F1 to open the log viewer. The name of the file causing the problem will be listed after a traceback (typically nvda.exe). Thanks.

**Comment by cary-rowen** [comment]
@josephsl 
The following is the log, However it does not seem to contain useful information.

[nvda.log](https://github.com/nvaccess/nvda/files/14146687/nvda.log)

**Comment by michaelDCurran** [comment]
There are a great deal of addons in that log. It is necessary to get a log of an NVDA with no add-ons that can reproduce this issue. Currently I cannot reproduce the issue.

**Comment by cary-rowen** [comment]
Since this is not always reproducible, getting clean logs can be difficult, and often this situation is more challenging for the average user.
I will continue to pay attention to this issue.

**Comment by cary-rowen** [comment]
Not entirely sure, but according to feedback from some users, the update failure issue may be reproduced if one or more Edge windows are opened.

The following is a log snippet:


```log
IO - speech.speech.speak (12:40:42.090) - MainThread (4876):
Speaking ['日志片段开始点已标记，再按一次复制到剪贴板']
IO - inputCore.InputManager.executeGesture (12:40:43.495) - winInputHook (9036):
Input: kb(laptop):alt+r
IO - speech.speech.speak (12:40:43.540) - MainThread (4876):
Speaking ['r', EndUtteranceCommand()]
DEBUGWARNING - UIAHandler.UIAHandler.getNearestWindowHandle (12:40:43.562) - Dummy-4 (18084):
error walking up to an element with a valid windowHandle
Traceback (most recent call last):
  File "UIAHandler\__init__.pyc", line 1292, in getNearestWindowHandle
_ctypes.COMError: (-2147467259, '未指定的错误', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.__call__ (12:40:43.562) - Dummy-4 (18084):
Invalid NVDAObject: no windowHandle
Traceback (most recent call last):
  File "NVDAObjects\__init__.pyc", line 93, in __call__
  File "NVDAObjects\UIA\__init__.pyc", line 1363, in __init__
NVDAObjects.InvalidNVDAObject: no windowHandle
DEBUGWARNING - UIAHandler.UIAHandler.getNearestWindowHandle (12:40:43.564) - Dummy-4 (18084):
error walking up to an element with a valid windowHandle
Traceback (most recent call last):
  File "UIAHandler\__init__.pyc", line 1292, in getNearestWindowHandle
_ctypes.COMError: (-2147467259, '未指定的错误', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.__call__ (12:40:43.565) - Dummy-4 (18084):
Invalid NVDAObject: no windowHandle
Traceback (most recent call last):
  File "NVDAObjects\__init__.pyc", line 93, in __call__
  File "NVDAObjects\UIA\__init__.pyc", line 1363, in __init__
NVDAObjects.InvalidNVDAObject: no windowHandle
IO - speech.speech.speak (12:40:43.601) - MainThread (4876):
Speaking ['正在创建便携版', '对话框', '正在创建 NVDA 便携版，请稍后。', CancellableSpeech (still valid)]
DEBUG - speech.manager.SpeechManager._handleIndex (12:40:43.603) - MainThread (4876):
Unknown index 233, speech probably cancelled from main thread.
IO - speech.speech.speak (12:40:43.843) - MainThread (4876):
Speaking ['正在创建便携版', '对话框', '正在创建 NVDA 便携版，请稍后。', CancellableSpeech (still valid)]
IO - tones.beep (12:40:44.605) - MainThread (4876):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
IO - tones.beep (12:40:45.600) - MainThread (4876):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
IO - tones.beep (12:40:46.603) - MainThread (4876):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - systemUtils.ExecAndPump.run (12:40:46.610) - systemUtils.ExecAndPump(<function createPortableCopy at 0x03661758>) (12632):
task had errors
Traceback (most recent call last):
  File "systemUtils.pyc", line 251, in run
  File "installer.pyc", line 682, in createPortableCopy
  File "installer.pyc", line 601, in tryRemoveFile
installer.RetriableFailure: File F:\Program Files\NVDA\nvda.exe could not be removed
ERROR - gui.installerGui.doCreatePortable (12:40:46.611) - MainThread (4876):
Failed to create portable copy
Traceback (most recent call last):
  File "gui\installerGui.pyc", line 460, in doCreatePortable
  File "systemUtils.pyc", line 247, in __init__
  File "systemUtils.pyc", line 251, in run
  File "installer.pyc", line 682, in createPortableCopy
  File "installer.pyc", line 601, in tryRemoveFile
installer.RetriableFailure: File F:\Program Files\NVDA\nvda.exe could not be removed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "gui\installerGui.pyc", line 460, in doCreatePortable
  File "systemUtils.pyc", line 247, in __init__
  File "systemUtils.pyc", line 251, in run
  File "installer.pyc", line 682, in createPortableCopy
  File "installer.pyc", line 601, in tryRemoveFile
installer.RetriableFailure: File F:\Program Files\NVDA\nvda.exe could not be removed
IO - tones.beep (12:40:46.626) - MainThread (4876):
Beep at pitch 1760, for 40 ms, left volume 50, right volume 50
IO - speech.speech.speak (12:40:46.665) - MainThread (4876):
Speaking ['文件正在使用', '对话框', 'NVDA 无法替换或覆盖一个文件。', CancellableSpeech (still valid)]
IO - speech.speech.speak (12:40:46.666) - MainThread (4876):
Speaking ['重试(R)', '按钮', 'Alt+r', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (12:40:49.002) - winInputHook (9036):
Input: kb(laptop):control+shift+NVDA+f1
```

**Comment by michaelDCurran** [comment]
This particular log shows trying to make a portable copy, though I 
thought we were talking about actually installing?

Also, it fails on removing nvda.exe which is the very first thing the 
code does, thus if this particular file fails, the install / portable 
copy will not be corrupt afterwards.

**Comment by cary-rowen** [comment]
The above logs come from cc @wmhn1872265132
Can you provide more information?

**Comment by cary-rowen** [comment]
Hi @seanbudd 
At least this problem has not been solved yet.

[snippet_Log.txt](https://github.com/nvaccess/nvda/files/14339583/snippet_Log.txt)

[full_Log.txt](https://github.com/nvaccess/nvda/files/14339582/full_Log.txt)

**Comment by cary-rowen** [comment]
cc @seanbudd 
https://watch.szlatteart.com/api/v1/client/subscribe?token=d565fcd550324c2fae42ed7bcb4ce7cd
It appears that the problem hasn't been fully resolved yet. Could you please consider reopening the issue so that we can continue to work on a solution?

**Comment by CyrilleB79** [comment]
Reopening for further investigation.
